src/libstore/local-store.cc

```diff
@@ -1320,7 +1321,18 @@ StorePath LocalStore::addToStoreFromDump(Source & source0, std::string_view name
 
     auto [hash, size] = hashSink->finish();
 
-    auto dstPath = makeFixedOutputPath(method, hash, name, references);
+    ContentAddressWithReferences desc = FixedOutputInfo {
+        .hash = {
+            .method = method,
+            .hash = hash,
+        },
+        .references = {
+            .others = references,
+            .self = false,
```

Comment:
```suggestion
            // caller is not capable of creating a self-reference, because this is content-addressed without modulus
            .self = false,
```